src/codegen.cpp

```diff
@@ -6344,6 +6346,158 @@ static void define_builtin_compile_vars(CodeGen *g) {
         }
         buf_appendf(contents, "};\n\n");
     }
+    {
+        // TODO: Add method info where methods are supported.
+        buf_appendf(contents,
```

Comment:
@andrewrk  
Will be done after all TypeInfo generation code is completed.

Reply:
@andrewrk  

I actually tried to implement the solution you suggested (all that was left was to get the type out of ir_analyze_instruction_field_ptr instead of getting the type of the type).  
However I found out that ir_analyze_instruction_field_ptr prioritizes the tag when accessing a union, so having the same struct names as the tags was not an option.  

I'm not sure if the current code is too hackish, it looks ok to me but I'm not acquainted to the compiler.

Reply:
Ah, I hadn't considered that. Hm, that sounds a bit like a design flaw. Hmmmmmmm. Oh well. It's really cosmetic. We can solve that problem later. What's important is the functionality of `@typeInfo` working.